### PR TITLE
OptionSelectorDelegate: avoid error when modelData is undefined and option.text===""

### PR DESCRIPTION
**Pull request by matdahl**
see issue #105

### CODE REVIEW

File: src/imports/Components/1.3/OptionSelectorDelegate.qml [hunk 1]
```diff
@@ -321,7 +321,7 @@ ListItem.Empty {
                 - parent.spacing * 2 - parent.anchors.leftMargin
 
             Toolkit.Label {
-                text: option.text === "" ? modelData : option.text
+                text: (option.text === "" && typeof modelData !== "undefined") ? modelData : option.text
```
Comment (dobey) [inline src/imports/Components/1.3/OptionSelectorDelegate.qml]: Why is `undefined` in quotes here?

**Review by dobey — CHANGES_REQUESTED** [review]

File: src/imports/Components/1.3/OptionSelectorDelegate.qml (hunk 1 above)
Comment (matdahl) [reply src/imports/Components/1.3/OptionSelectorDelegate.qml]: because 
```js
typeof modelData
```
is a string

Comment (dobey) [reply src/imports/Components/1.3/OptionSelectorDelegate.qml]: Why are you not just doing `modelData !== undefined` though, as pretty much all the other code checking against undefined, does?

**Comment by dobey** [comment]
Also can you please do a `git pull --rebase origin xenial` and then `git push --force` in this branch? It seems the build is failing due to being out of sync with the changes that landed for supporting Qt 5.12, and CI doesn't do a three way merge when building.

File: src/imports/Components/1.3/OptionSelectorDelegate.qml (hunk 1 above)
Comment (matdahl) [reply src/imports/Components/1.3/OptionSelectorDelegate.qml]: when just writing `modelData !== undefined` I get again a
```
ReferenceError: modelData is not defined
```

Comment (matdahl) [reply src/imports/Components/1.3/OptionSelectorDelegate.qml]: I got the solution from here:
https://askubuntu.com/questions/527799/how-do-you-check-if-a-property-is-undefined-in-qml/552586#552586

**Comment by matdahl** [comment]
> Also can you please do a `git pull --rebase origin xenial` and then `git push --force` in this branch? It seems the build is failing due to being out of sync with the changes that landed for supporting Qt 5.12, and CI doesn't do a three way merge when building.

I did it. But it seems like nothing happend ...

**Comment by dobey** [comment]
> I did it. But it seems like nothing happend ...

Ah, sorry. I guess your `origin` is not the upstream, but your forked repo. You'll need to do `git remote add upstream https://github.com/ubports/ubuntu-ui-toolkit.git` and then do `git pull --rebase upstream xenial` to pull in the commits for Qt 5.12 support. Then push that to your repo. Thanks.

**Comment by matdahl** [comment]
> > I did it. But it seems like nothing happend ...
> 
> Ah, sorry. I guess your `origin` is not the upstream, but your forked repo. You'll need to do `git remote add upstream https://github.com/ubports/ubuntu-ui-toolkit.git` and then do `git pull --rebase upstream xenial` to pull in the commits for Qt 5.12 support. Then push that to your repo. Thanks.

Thanks! Now it worked

File: src/imports/Components/1.3/OptionSelectorDelegate.qml (hunk 1 above)
Comment (dobey) [reply src/imports/Components/1.3/OptionSelectorDelegate.qml]: Can you please change it to just do `&& modelData !== undefined` instead? Thanks.

Comment (matdahl) [reply src/imports/Components/1.3/OptionSelectorDelegate.qml]: I changed it although on my system, the `modelData !== undefined` still throws the same ReferenceError as the current code.

Comment (dobey) [reply src/imports/Components/1.3/OptionSelectorDelegate.qml]: Hmm. Is there a specific app or such which shows that? I don't see it occurring in the test output during the build. Perhaps we could add a test which shows the problem, so we can prevent it from happening again?

Comment (matdahl) [reply src/imports/Components/1.3/OptionSelectorDelegate.qml]: This problem is showing up in all my self developed apps. Each time an OptionSelectorDelegate instance is being created. Interestingly, the above linked answer on askubuntu describes exactly what I find. Maybe it's a bug that's fixed with Qt 5.12?

Comment (dobey) [reply src/imports/Components/1.3/OptionSelectorDelegate.qml]: Not sure. I don't see a ReferenceError here with UI Toolkit Gallery app either. I still keep seeing the binding loop issue though, even with change here.

```
file:///usr/lib/arm-linux-gnueabihf/qt5/qml/Ubuntu/Components/1.3/OptionSelector.qml:330:13: QML ListView: Binding loop detected for property "itemHeight"
```

Comment (dobey) [reply src/imports/Components/1.3/OptionSelectorDelegate.qml]: I'm inclined to close this for now, as it's not clear what's going on, and I can't reproduce the given issue. Can you attach a minimal example which shows the issue happening, in the #105 ticket? We're trying to get the repo migrated to GitLab as we work on updating to 20.04 (and this one is lower in the dependency stack), and I'd like to reproduce the issue and try to understand why using `typeof` and comparing the string would work but checking against `undefined` would not before approving.

Thanks!

**Comment by dobey** [comment]
OK. Per my previous comment in the thread above, I'm going to close this for now. We can revisit after the migration to GitLab. Thanks!

**Pull request closed without merging**